lib/peer.ml

```diff
@@ -24,3 +24,36 @@ let from_sockaddr sockaddr =
 
 let to_sockaddr peer =
   Lwt_unix.ADDR_INET (Unix.inet_addr_of_string peer.address, peer.port)
+
+(* Basic Knuth shuffle => https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle *)
+let knuth_shuffle known_peers =
+  let initial_array = Array.length (Array.of_list known_peers) in
+  let shuffled_array = Array.copy (Array.of_list known_peers) in
+  for i = initial_array - 1 downto 1 do
```

Comment:
Is there any `functional` way to write it?

Reply:
Define a swap function on indices i and j, then Array.iteri to swap each index i with a random index j?